Sources/FoundationEssentials/Error/CocoaError+FilePath.swift

```diff
@@ -129,6 +129,18 @@ extension CocoaError {
         )
     }
 
+    static func errorWithFilePath(_ code: CocoaError.Code, _ path: String? = nil, variant: String? = nil, userInfo: [String : AnyHashable] = [:]) -> CocoaError {
+        var info: [String : AnyHashable] = userInfo
+        if let path, info[NSFilePathErrorKey] == nil {
```

Comment:
Sorry to nitpick on this, but instead of setting `NSFilePathErrorKey` directly here, could we use `userInfo.addingUserInfo(forPath: path)`? This will also take care of adding `NSURLErrorKey` and will help minimize duplicated code. Otherwise this changes looks good to me